src/utils.rs

```diff
@@ -51,6 +51,41 @@ pub fn snippet<'a>(cx: &Context, span: Span, default: &'a str) -> Cow<'a, str> {
     cx.sess().codemap().span_to_snippet(span).map(From::from).unwrap_or(Cow::Borrowed(default))
 }
 
+/// convert a span (from a block) to a code snippet if available, otherwise use default, e.g.
+/// `snippet(cx, expr.span, "..")`
+/// This trims the code of indentation, but in some cases
+/// the first line has no indentation
+pub fn snippet_block<'a>(cx: &Context, span: Span, default: &'a str) -> Cow<'a, str> {
+    let snip = snippet(cx, span, default);
+    trim_multiline(snip, true)
+}
+
+/// Trim indentation from a multiline string
+/// with possibility of ignoring the first line
+pub fn trim_multiline<'a>(s: Cow<'a, str>, ignore_first: bool) -> Cow<'a, str> {
+    let skip = if ignore_first {
+        1
+    } else {
+        0
+    };
+    let x = s.lines().skip(skip)
+             .map(|l| l.char_indices()
+                       .find(|&(_,x)| x != ' ')
+                       .unwrap_or((l.len(),' ')).0)
+             .min().unwrap_or(0);
+    if x > 0 {
+        let mut ignored = !ignore_first;
```

Comment:
I tried that and it was uglier, but the code has changed a lot since then so it could probably be done better. Will do.